### ISSUE TITLE
Which version is more correct? Please see the results I produced?

### ISSUE THREAD

**Issue by alitekdemir** [freqtrade/freqtrade#6525]
## My environment

  * Operating system: Windows 10 Pro
  * Python Version: Python 3.10.2
  * CCXT version: ccxt==1.72.98
  * Freqtrade Version:  2022.2.1

While the "backtest" results I did with freqtrade v2021.1 looked much more profitable, this was completely reversed with the new version!

- I started using limit prices while testing my strategy.
- I try to generate results with --timeframe-detail 1m
- I managed to make progress in the results, but everything changed with the new version.
- Maybe this new version is more accurate I don't know
- The number of buys varies a lot.

## This is good result with v2022.1

```
(.env) PS C:\freqtrade> pip install freqtrade==2022.1
(.env) PS C:\freqtrade> freqtrade -V
freqtrade 2022.1
(.env) PS C:\freqtrade> freqtrade backtesting --timerange 20210601-20220101 --timeframe 15m --timeframe-detail 1m -s ATDv1 -p ETH/USDT

Result for strategy ATDv1
========================================================== BACKTESTING REPORT =========================================================
|     Pair |   Buys |   Avg Profit % |   Cum Profit % |   Tot Profit USDT |   Tot Profit % |   Avg Duration |   Win  Draw  Loss  Win% |
|----------+--------+----------------+----------------+-------------------+----------------+----------------+-------------------------|
| ETH/USDT |    178 |           0.59 |         104.31 |          1655.075 |         165.51 |        8:04:00 |    81     0    97  45.5 |
|    TOTAL |    178 |           0.59 |         104.31 |          1655.075 |         165.51 |        8:04:00 |    81     0    97  45.5 |
========================================================== BUY TAG STATS ===========================================================
|   TAG |   Buys |   Avg Profit % |   Cum Profit % |   Tot Profit USDT |   Tot Profit % |   Avg Duration |   Win  Draw  Loss  Win% |
|-------+--------+----------------+----------------+-------------------+----------------+----------------+-------------------------|
| TOTAL |    178 |           0.59 |         104.31 |          1655.075 |         165.51 |        8:04:00 |    81     0    97  45.5 |
===================================================== SELL REASON STATS =====================================================
|   Sell Reason |   Sells |   Win  Draws  Loss  Win% |   Avg Profit % |   Cum Profit % |   Tot Profit USDT |   Tot Profit % |
|---------------+---------+--------------------------+----------------+----------------+-------------------+----------------|
|   sell_signal |     176 |     79     0    97  44.9 |           0.48 |          84.33 |          1377.49  |          84.33 |
|           roi |       2 |      2     0     0   100 |           9.99 |          19.98 |           277.581 |          19.98 |
====================================================== LEFT OPEN TRADES REPORT ======================================================
|   Pair |   Buys |   Avg Profit % |   Cum Profit % |   Tot Profit USDT |   Tot Profit % |   Avg Duration |   Win  Draw  Loss  Win% |
|--------+--------+----------------+----------------+-------------------+----------------+----------------+-------------------------|
|  TOTAL |      0 |           0.00 |           0.00 |             0.000 |           0.00 |           0:00 |     0     0     0     0 |
=============== SUMMARY METRICS ================
| Metric                 | Value               |
|------------------------+---------------------|
| Backtesting from       | 2021-06-01 00:00:00 |
| Backtesting to         | 2022-01-01 00:00:00 |
| Max open trades        | 1                   |
|                        |                     |
| Total/Daily Avg Trades | 178 / 0.83          |
| Starting balance       | 1000.000 USDT       |
| Final balance          | 2655.075 USDT       |
| Absolute profit        | 1655.075 USDT       |
| Total profit %         | 165.51%             |
| Trades per day         | 0.83                |
| Avg. daily profit %    | 0.77%               |
| Avg. stake amount      | 1795.491 USDT       |
| Total trade volume     | 319597.471 USDT     |
|                        |                     |
| Best Pair              | ETH/USDT 104.31%    |
| Worst Pair             | ETH/USDT 104.31%    |
| Best trade             | ETH/USDT 9.99%      |
| Worst trade            | ETH/USDT -3.34%     |
| Best day               | 199.165 USDT        |
| Worst day              | -103.938 USDT       |
| Days win/draw/lose     | 68 / 78 / 68        |
| Avg. Duration Winners  | 12:40:00            |
| Avg. Duration Loser    | 4:14:00             |
| Rejected Buy signals   | 0                   |
|                        |                     |
| Min balance            | 965.884 USDT        |
| Max balance            | 2666.171 USDT       |
| Drawdown (Account)     | 7.30%               |
| Drawdown               | 179.825 USDT        |
| Drawdown high          | 1462.988 USDT       |
| Drawdown low           | 1283.163 USDT       |
| Drawdown Start         | 2021-10-14 21:15:00 |
| Drawdown End           | 2021-11-05 09:45:00 |
| Market change          | 44.44%              |
================================================
```

## This is new version results
```
freqtrade backtesting --timerange 20210601-20220101 --timeframe 15m -s ATDv1 -p ETH/USDT
(.env) PS C:\freqtrade> pip install --upgrade freqtrade
(.env) PS C:\freqtrade> freqtrade -V
freqtrade 2022.2.1
(.env) PS C:\freqtrade> freqtrade backtesting --timerange 20210601-20220101 --timeframe 15m -s ATDv1 -p ETH/USDT

Result for strategy ATDv1
========================================================== BACKTESTING REPORT ==========================================================
|     Pair |   Buys |   Avg Profit % |   Cum Profit % |   Tot Profit USDT |   Tot Profit % |    Avg Duration |   Win  Draw  Loss  Win% |
|----------+--------+----------------+----------------+-------------------+----------------+-----------------+-------------------------|
| ETH/USDT |     80 |           0.16 |          12.42 |            -4.448 |          -0.44 | 1 day, 16:31:00 |    30     0    50  37.5 |
|    TOTAL |     80 |           0.16 |          12.42 |            -4.448 |          -0.44 | 1 day, 16:31:00 |    30     0    50  37.5 |
=========================================================== BUY TAG STATS ===========================================================
|   TAG |   Buys |   Avg Profit % |   Cum Profit % |   Tot Profit USDT |   Tot Profit % |    Avg Duration |   Win  Draw  Loss  Win% |
|-------+--------+----------------+----------------+-------------------+----------------+-----------------+-------------------------|
| TOTAL |     80 |           0.16 |          12.42 |            -4.448 |          -0.44 | 1 day, 16:31:00 |    30     0    50  37.5 |
===================================================== SELL REASON STATS =====================================================
|   Sell Reason |   Sells |   Win  Draws  Loss  Win% |   Avg Profit % |   Cum Profit % |   Tot Profit USDT |   Tot Profit % |
|---------------+---------+--------------------------+----------------+----------------+-------------------+----------------|
|     stop_loss |      44 |      0     0    44     0 |          -4.2  |        -184.71 |         -1703.31  |        -184.71 |
|           roi |      18 |     18     0     0   100 |           9.99 |         179.82 |          1541.84  |         179.82 |
|   sell_signal |      17 |     12     0     5  70.6 |           1.09 |          18.49 |           168.881 |          18.49 |
|    force_sell |       1 |      0     0     1     0 |          -1.19 |          -1.19 |           -11.864 |          -1.19 |
======================================================= LEFT OPEN TRADES REPORT ========================================================
|     Pair |   Buys |   Avg Profit % |   Cum Profit % |   Tot Profit USDT |   Tot Profit % |    Avg Duration |   Win  Draw  Loss  Win% |
|----------+--------+----------------+----------------+-------------------+----------------+-----------------+-------------------------|
| ETH/USDT |      1 |          -1.19 |          -1.19 |           -11.864 |          -1.19 | 1 day, 15:45:00 |     0     0     1     0 |
|    TOTAL |      1 |          -1.19 |          -1.19 |           -11.864 |          -1.19 | 1 day, 15:45:00 |     0     0     1     0 |
=============== SUMMARY METRICS ================
| Metric                 | Value               |
|------------------------+---------------------|
| Backtesting from       | 2021-06-01 00:00:00 |
| Backtesting to         | 2022-01-01 00:00:00 |
| Max open trades        | 1                   |
|                        |                     |
| Total/Daily Avg Trades | 80 / 0.37           |
| Starting balance       | 1000 USDT           |
| Final balance          | 995.552 USDT        |
| Absolute profit        | -4.448 USDT         |
| Total profit %         | -0.44%              |
| Trades per day         | 0.37                |
| Avg. daily profit %    | -0.00%              |
| Avg. stake amount      | 896.581 USDT        |
| Total trade volume     | 71726.463 USDT      |
|                        |                     |
| Best Pair              | ETH/USDT 12.42%     |
| Worst Pair             | ETH/USDT 12.42%     |
| Best trade             | ETH/USDT 9.99%      |
| Worst trade            | ETH/USDT -4.46%     |
| Best day               | 103.072 USDT        |
| Worst day              | -89.951 USDT        |
| Days win/draw/lose     | 29 / 144 / 42       |
| Avg. Duration Winners  | 2 days, 16:53:00    |
| Avg. Duration Loser    | 1 day, 1:54:00      |
| Rejected Buy signals   | 0                   |
| Entry/Exit Timeouts    | 35 / 374            |
|                        |                     |
| Min balance            | 598.061 USDT        |
| Max balance            | 1144.2 USDT         |
| Drawdown (Account)     | 42.80%              |
| Drawdown               | 447.482 USDT        |
| Drawdown high          | 45.543 USDT         |
| Drawdown low           | -401.939 USDT       |
| Drawdown Start         | 2021-06-14 13:30:00 |
| Drawdown End           | 2021-07-19 13:00:00 |
| Market change          | 44.44%              |
================================================
```

## If I add a minute of detail, the result doesn't change much.

```
(.env) PS C:\freqtrade> freqtrade backtesting --timerange 20210601-20220101 --timeframe 15m --timeframe-detail 1m -s ATDv1 -p ETH/USDT

Result for strategy ATDv1
========================================================== BACKTESTING REPORT ==========================================================
|     Pair |   Buys |   Avg Profit % |   Cum Profit % |   Tot Profit USDT |   Tot Profit % |    Avg Duration |   Win  Draw  Loss  Win% |
|----------+--------+----------------+----------------+-------------------+----------------+-----------------+-------------------------|
| ETH/USDT |     80 |           0.14 |          11.13 |           -17.603 |          -1.76 | 1 day, 16:31:00 |    30     0    50  37.5 |
|    TOTAL |     80 |           0.14 |          11.13 |           -17.603 |          -1.76 | 1 day, 16:31:00 |    30     0    50  37.5 |
=========================================================== BUY TAG STATS ===========================================================
|   TAG |   Buys |   Avg Profit % |   Cum Profit % |   Tot Profit USDT |   Tot Profit % |    Avg Duration |   Win  Draw  Loss  Win% |
|-------+--------+----------------+----------------+-------------------+----------------+-----------------+-------------------------|
| TOTAL |     80 |           0.14 |          11.13 |           -17.603 |          -1.76 | 1 day, 16:31:00 |    30     0    50  37.5 |
===================================================== SELL REASON STATS =====================================================
|   Sell Reason |   Sells |   Win  Draws  Loss  Win% |   Avg Profit % |   Cum Profit % |   Tot Profit USDT |   Tot Profit % |
|---------------+---------+--------------------------+----------------+----------------+-------------------+----------------|
|     stop_loss |      44 |      0     0    44     0 |          -4.23 |        -185.99 |         -1701.38  |        -185.99 |
|           roi |      18 |     18     0     0   100 |           9.99 |         179.82 |          1528.94  |         179.82 |
|   sell_signal |      17 |     12     0     5  70.6 |           1.09 |          18.49 |           166.538 |          18.49 |
|    force_sell |       1 |      0     0     1     0 |          -1.19 |          -1.19 |           -11.707 |          -1.19 |
======================================================= LEFT OPEN TRADES REPORT ========================================================
|     Pair |   Buys |   Avg Profit % |   Cum Profit % |   Tot Profit USDT |   Tot Profit % |    Avg Duration |   Win  Draw  Loss  Win% |
|----------+--------+----------------+----------------+-------------------+----------------+-----------------+-------------------------|
| ETH/USDT |      1 |          -1.19 |          -1.19 |           -11.707 |          -1.17 | 1 day, 15:45:00 |     0     0     1     0 |
|    TOTAL |      1 |          -1.19 |          -1.19 |           -11.707 |          -1.17 | 1 day, 15:45:00 |     0     0     1     0 |
=============== SUMMARY METRICS ================
| Metric                 | Value               |
|------------------------+---------------------|
| Backtesting from       | 2021-06-01 00:00:00 |
| Backtesting to         | 2022-01-01 00:00:00 |
| Max open trades        | 1                   |
|                        |                     |
| Total/Daily Avg Trades | 80 / 0.37           |
| Starting balance       | 1000 USDT           |
| Final balance          | 982.397 USDT        |
| Absolute profit        | -17.603 USDT        |
| Total profit %         | -1.76%              |
| Trades per day         | 0.37                |
| Avg. daily profit %    | -0.01%              |
| Avg. stake amount      | 889.662 USDT        |
| Total trade volume     | 71172.939 USDT      |
|                        |                     |
| Best Pair              | ETH/USDT 11.13%     |
| Worst Pair             | ETH/USDT 11.13%     |
| Best trade             | ETH/USDT 9.99%      |
| Worst trade            | ETH/USDT -4.79%     |
| Best day               | 101.71 USDT         |
| Worst day              | -89.951 USDT        |
| Days win/draw/lose     | 29 / 144 / 42       |
| Avg. Duration Winners  | 2 days, 16:53:00    |
| Avg. Duration Loser    | 1 day, 1:54:00      |
| Rejected Buy signals   | 0                   |
| Entry/Exit Timeouts    | 35 / 374            |
|                        |                     |
| Min balance            | 593.861 USDT        |
| Max balance            | 1129.081 USDT       |
| Drawdown (Account)     | 43.20%              |
| Drawdown               | 451.682 USDT        |
| Drawdown high          | 45.543 USDT         |
| Drawdown low           | -406.139 USDT       |
| Drawdown Start         | 2021-06-14 13:30:00 |
| Drawdown End           | 2021-07-19 13:00:00 |
| Market change          | 44.44%              |
================================================

```

## If i don't use limit price, result is change to this
```
freqtrade backtesting --timerange 20210601-20220101 --timeframe 15m -s ATDv1 -p ETH/USDT

========================================================== BACKTESTING REPORT =========================================================
|     Pair |   Buys |   Avg Profit % |   Cum Profit % |   Tot Profit USDT |   Tot Profit % |   Avg Duration |   Win  Draw  Loss  Win% |
|----------+--------+----------------+----------------+-------------------+----------------+----------------+-------------------------|
| ETH/USDT |    178 |           0.19 |          34.21 |           329.353 |          32.94 |        8:04:00 |    79     0    99  44.4 |
|    TOTAL |    178 |           0.19 |          34.21 |           329.353 |          32.94 |        8:04:00 |    79     0    99  44.4 |
========================================================== BUY TAG STATS ===========================================================
|   TAG |   Buys |   Avg Profit % |   Cum Profit % |   Tot Profit USDT |   Tot Profit % |   Avg Duration |   Win  Draw  Loss  Win% |
|-------+--------+----------------+----------------+-------------------+----------------+----------------+-------------------------|
| TOTAL |    178 |           0.19 |          34.21 |           329.353 |          32.94 |        8:04:00 |    79     0    99  44.4 |
===================================================== SELL REASON STATS =====================================================
|   Sell Reason |   Sells |   Win  Draws  Loss  Win% |   Avg Profit % |   Cum Profit % |   Tot Profit USDT |   Tot Profit % |
|---------------+---------+--------------------------+----------------+----------------+-------------------+----------------|
|   sell_signal |     175 |     77     0    98  44.0 |           0.11 |          18.42 |           155.791 |          18.42 |
|           roi |       2 |      2     0     0   100 |           9.99 |          19.98 |           215.7   |          19.98 |
|     stop_loss |       1 |      0     0     1     0 |          -4.19 |          -4.19 |           -42.138 |          -4.19 |
====================================================== LEFT OPEN TRADES REPORT ======================================================
|   Pair |   Buys |   Avg Profit % |   Cum Profit % |   Tot Profit USDT |   Tot Profit % |   Avg Duration |   Win  Draw  Loss  Win% |
|--------+--------+----------------+----------------+-------------------+----------------+----------------+-------------------------|
|  TOTAL |      0 |           0.00 |           0.00 |             0.000 |           0.00 |           0:00 |     0     0     0     0 |
=============== SUMMARY METRICS ================
| Metric                 | Value               |
|------------------------+---------------------|
| Backtesting from       | 2021-06-01 00:00:00 |
| Backtesting to         | 2022-01-01 00:00:00 |
| Max open trades        | 1                   |
|                        |                     |
| Total/Daily Avg Trades | 178 / 0.83          |
| Starting balance       | 1000 USDT           |
| Final balance          | 1329.353 USDT       |
| Absolute profit        | 329.353 USDT        |
| Total profit %         | 32.94%              |
| Trades per day         | 0.83                |
| Avg. daily profit %    | 0.15%               |
| Avg. stake amount      | 1209.053 USDT       |
| Total trade volume     | 215211.391 USDT     |
|                        |                     |
| Best Pair              | ETH/USDT 34.21%     |
| Worst Pair             | ETH/USDT 34.21%     |
| Best trade             | ETH/USDT 9.99%      |
| Worst trade            | ETH/USDT -4.19%     |
| Best day               | 142.223 USDT        |
| Worst day              | -63.362 USDT        |
| Days win/draw/lose     | 61 / 78 / 75        |
| Avg. Duration Winners  | 12:58:00            |
| Avg. Duration Loser    | 4:10:00             |
| Rejected Buy signals   | 0                   |
| Entry/Exit Timeouts    | 0 / 0               |
|                        |                     |
| Min balance            | 941.688 USDT        |
| Max balance            | 1459.387 USDT       |
| Drawdown (Account)     | 14.82%              |
| Drawdown               | 216.213 USDT        |
| Drawdown high          | 459.387 USDT        |
| Drawdown low           | 243.174 USDT        |
| Drawdown Start         | 2021-10-14 21:15:00 |
| Drawdown End           | 2021-12-04 00:45:00 |
| Market change          | 44.44%              |
================================================
```

**Comment by xmatthias** [comment]
The point you seemingly missed is this:
```
| Entry/Exit Timeouts    | 35 / 374            |
```

> Timeout handling - this will change how custom-pricing works in backtesting. check_buy_timeout() and check_sell_timeout() are now called in backtesting for orders with prices outside of the opening candle.

It's a point we announced in the [last release](https://github.com/freqtrade/freqtrade/releases/tag/2022.2.1) 

As you seem to be using custom pricing (with prices that can be outside of the candle) - you're encountering many missed exits - and "a few" missed entries.
Prior behavior was to move this price to be within the opening/closing candle (within high/low range) - so the price your strategy assigned was slightly modified, and accepted as is.

If we look at the exit timeouts (which there are quite many of) - you can assume that you placed your orders 1-2% above current high.
Now in reality - if you're lucky - price will reach this - and the order get's filled.
More likely - the order will time out.
Backtesting did move this "price from 2% above high" to high - and assumed you made a perfect sell (exactly selling at the high point) - which is obviously quite unlikely.
The same did happen on entries, too - where price was moved to "low" - making a perfect entry again.

So in reality / live trading, this adjustment will not happen (we won't know "high" before it happens - and then it's too late to place the order) - and the order would stay there until it times out (price moved in the other direction) (you most likely encountered this in dry/live if you run this strategy in live).

Now (new version) backtesting will account for this to the best of it's abilities (we still only have 4 prices per candle - but we know if a price is within the candle or not) - and can therefore determine that your prices were "away from current price" - and led to "this many" sells.

In short - your price adjustments `custom_*_price` callbacks are too aggressive - and therefore did exploit a known behavior (we had it documented this way - with the note that it's a point to be careful about).

Market pricing will ignore your "price suggestions" - as market prices fill immediately at "current price" - therefore this will (kindof) restore prior behavior of all orders filling immediately.

---

The number of trades will obviously vary - as you had 34 trades which were not entered - and a few 100 missed exits (so your trade duration is longer - therefore no new trades can be made on this one pair).
